woocommerce/woocommerce-android
# Tests: Remove unused UI instrumentation tests and mocking

## Changes

As discussed (internal ref: p91TBi-348-p2#comment-1332) this PR removes our mostly unused UI instrumentation tests. Future work will expand the approach used in our remaining Screenshot Test and add supporting infrastructure, to focus on testing full user flows/journeys (things like navigation, user interactions, successful outcomes) rather than using instrumentation tests to check data in the UI.

A few notes about this PR:

* I left some things in place (like `/di/AppComponentTest` and `/helpers/TestBase`) to reuse in the new network mocking and test infrastructure.
* I left the helpers (custom matchers, utils) for now, since they could be useful in new tests. There may be some duplication between those helpers and the ones in `/screenshots/util` that we can look at cleaning up in a future PR.
* With the changes, the remaining test in `/screenshots/ScreenshotTest` will build and run but fail. It needs to be updated to use stats v4 in order to pass; we can fix that test in a followup PR.

## Testing

To test:

* Confirm the instrumentation tests build (`./gradlew assembleVanillaDebugAndroidTest`).

## Review

This PR should only need one reviewer, and anyone can review it. If you see anything that was removed that you think shouldn't be, or anything that wasn't removed but should be, please flag it.

## Submitter checklist

Update release notes:
